src/daily_output.test.ts

```diff
@@ -0,0 +1,170 @@
+import { expect } from 'chai';
+import { rmSync, writeFileSync } from 'fs';
+import Sinon from 'sinon';
+import { OUTPUT_NAME } from './constants';
+import { DailyOutput } from './daily_output';
+
+// this JSON is different than the base template
+const mockDailyOutput = {
+	blockHeight: 914200,
+	timestamp: 1650043449000,
+	PSTHolders: {},
+	wallets: {},
+	ranks: {
+		daily: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		weekly: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		lastWeek: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		total: {
+			groupEffortRewards: [],
+			streakRewards: []
+		}
+	}
+};
+
+// The avobe JSON stringified with tabs and a trailing new line
+const mockDailyOutputStringified = `${JSON.stringify(mockDailyOutput, null, '\t')}\n`;
+
+// a valid JSON with missing mandatory fields
+const mockMalformedDailyOutput = {
+	blockHeight: 914200,
+	timestamp: 1650043449000,
+	PSTHolders: {},
+	// wallets: {},
+	ranks: {
+		daily: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		weekly: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		},
+		lastWeek: {
+			hasReachedMinimumGroupEffort: false,
+			groupEffortRewards: [],
+			streakRewards: []
+		}
+		// total: {
+		// 	groupEffortRewards: [],
+		// 	streakRewards: []
+		// }
+	}
+};
+
+// The avobe JSON stringified with tabs and a trailing new line
+const mockMalformedDailyOutputStringified = `${JSON.stringify(mockMalformedDailyOutput, null, '\t')}\n`;
+
+describe('DailyOutput class', () => {
+	const output = new DailyOutput();
+
+	describe('read method', () => {
+		before(() => {
+			rmSync(OUTPUT_NAME);
+		});
```

Comment:
This is breaking in my environment. I don't have this output file on install

Reply:
15dc2ab1818c82727fee93cd857281adbdcafacd